src/test/kotlin/LoxoneAT.kt

```diff
@@ -0,0 +1,188 @@
+package cz.smarteon.loxone
+
+import cz.smarteon.loxone.app.SwitchControl
+import cz.smarteon.loxone.message.JsonValue
+import cz.smarteon.loxone.message.LoxoneMessage
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.MethodOrderer
+import org.junit.jupiter.api.Order
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestMethodOrder
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable
+import strikt.api.expectCatching
+import strikt.api.expectThat
+import strikt.assertions.hasSize
+import strikt.assertions.isA
+import strikt.assertions.isEqualTo
+import strikt.assertions.isNotEqualTo
+import strikt.assertions.isSuccess
+import strikt.assertions.isTrue
+import java.security.Security
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import kotlin.collections.LinkedHashMap
+import kotlin.collections.first
+import kotlin.collections.set
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation::class)
+@EnabledIfEnvironmentVariable(named = LoxoneAT.LOX_ADDRESS, matches = ".*")
+@EnabledIfEnvironmentVariable(named = LoxoneAT.LOX_USER, matches = ".*")
+@EnabledIfEnvironmentVariable(named = LoxoneAT.LOX_PASS, matches = ".*")
+@EnabledIfEnvironmentVariable(named = LoxoneAT.LOX_VISPASS, matches = ".*")
+class LoxoneAT {
+    private lateinit var loxone: Loxone
+    private lateinit var commands: CommandResponseMemory
+    private lateinit var device: SwitchControl
+    private lateinit var secDevice: SwitchControl
+
+    @BeforeAll
+    fun setup() {
+        Security.addProvider((BouncyCastleProvider()))
+
+        val port = if (System.getenv(LOX_PORT) != null) System.getenv(LOX_PORT).toInt() else 80
+        val useSsl = if (System.getenv(LOX_SSL) != null) System.getenv(LOX_SSL).toBoolean() else false
+        val endpoint = LoxoneEndpoint(System.getenv(LOX_ADDRESS), port, useSsl)
+        loxone = Loxone(endpoint, System.getenv(LOX_USER), System.getenv(LOX_PASS), System.getenv(LOX_VISPASS))
+
+        commands = CommandResponseMemory()
+        loxone.webSocket().registerListener(commands)
+
+        loxone.start()
+
+        device = loxone.app()?.getControl(DEVICE_NAME, SwitchControl::class.java)!!
```

Comment:
in case there is no such device NullPointer will be raised here :(

1. make both devices nullable
2. assert them properly in `should have testing devices`